posydon/binary_evol/DT/step_detached.py

```diff
@@ -599,188 +621,91 @@ def get_mist0(self, star, htrack):
-                    + (sc_center_he4_H.transform(
-                        get_track_val(MESA_label[2], htrack, *x))
-                       - sc_center_he4_H.transform(posydon_attribute[2])) ** 2
-                    + (sc_log_R_H.transform(
-                        get_track_val(MESA_label[3], htrack, *x))
-                        - sc_log_R_H.transform(posydon_attribute[3])) ** 2,
-
-                    x0, method="TNC", bounds=bnds,
-                    )
-            elif star.state in LIST_ACCEPTABLE_STATES_FOR_HeStar:
-                MESA_label = [
-                    "he_core_mass",
-                    # "surface_he4",
-                    "center_he4",
-                    # "mass",
-                    # "center_c12",
-                    "log_R"
-                ]
-                posydon_attribute = [
-                    star.he_core_mass,
-                    # star.surface_he4,
-                    star.center_he4,
-                    # star.mass,
-                    # star.center_c12,
-                    star.log_R
-                ]
-                # rs = [10, 2.0, 2.0, 20, 2.0]
-                rs = [10.0, 1.0, 2.0]
-                if self.verbose:
-                    print("Matching attributes and their normalizations : ",
-                          MESA_label, rs)
-                for i in MESA_label:
-                    if i not in self.root_keys:
-                        raise Exception("Expected matching parameter not added"
-                                        " in the MIST model options.")
-                x0 = get_root0(MESA_label, posydon_attribute, False, rs=rs)
-                bnds = ([m_min_He, m_max_He], [0, None])
-                # match he4 abundance using definite value
-                sol = minimize(
-                    lambda x: (
-                        (sc_he_core_mass_He.transform(
-                         get_track_val(MESA_label[0], htrack, *x))
-                         - sc_he_core_mass_He.transform(posydon_attribute[0]))
-                        / sc_he_core_mass_He.transform(posydon_attribute[0]))
-                    ** 2 + (sc_center_he4_He.transform(
-                        get_track_val(MESA_label[1], htrack, *x))
-                            - sc_center_he4_He.transform(posydon_attribute[1]))
-                    ** 2 + (sc_log_R_He.transform(
-                        get_track_val(MESA_label[2], htrack, *x))
-                            - sc_log_R_He.transform(posydon_attribute[2]))**2,
-                    x0, method="TNC", bounds=bnds,
-                )
-                if (np.abs(sol.fun) > tolerance_mist_integration
+            '''
+            ####This is for He-star 
+            '''
+                if (np.abs(sol.fun) > tolerance_matching_integration
                         or not sol.success):
-                    sol = minimize(
-                        lambda x: (
-                            (sc_he_core_mass_He.transform(
```

Comment:
Hello @ZepeiX . With @kasdaglie we need you input on this change of trying to slowly implement a step of stars that are doing an isolated evolution (initially single stars, disrupted binaries or mergers) based on the detached step. For now we are only focusing on the disrupted binaries.

We tried to generalize the detached step to accept as an input the parameters that will be used for the matching, and not hard coded. We saw that you had some alternative matching trials if the first match is not working. I know that you have tested it and you have come up with them as optimal at least at that point. For now we do not do it, we only try to do the first matching trial that was there. Do you think we should implement the alternative ones? Specifically, we saw that you sometime change from matching the absolute mass to the relative mass, as in this case (dividing by the mass value itself). Can you suggest which of these to keep and why?

Reply:
Now the known issues of failed matching include:
1. MS stars after MT has a smaller radius so that we match it again without radius
2. An almost stripped post-MS star cannot be matched within the single H-grid while it is more like a He star and we match it to the He-grid
3. Some He stars cannot be matched within single He-grid because when they got stripped, they have evolved a bit inside which cannot be reproduced with He-MS. But we found they are more similar to evolved H stars so that we match them in H-grid.